### PR TITLE
Add PL PESEL recognizer

### PR DESCRIPTION
**Pull request by bckamil**
## Change Description

Add a recognizer for the Polish PESEL number.

## Checklist

- [x] I have reviewed the [contribution guidelines](https://github.com/microsoft/presidio/blob/main/CONTRIBUTING.md)
- [x] I have signed the CLA (if required)
- [x] My code includes unit tests
- [x] All unit tests and lint checks pass locally
- [x] My PR contains documentation updates / additions if required

### CODE REVIEW

**Comment by bckamil** [comment]
@microsoft-github-policy-service agree

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    :param patterns: List of patterns to be used by this recognizer
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS = [
+        Pattern(
+            "PESEL",
+            r"^[0-9]{2}([02468][1-9]|[13579][012])"
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py:21]: Could you please provide the references to these regex patterns and the validation code?

Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py:21]: I tried them on what seems to be a valid PESEL (I'm certainly not an expert so maybe I got it wrong) but it only detected part of the pesel text.

**Review by omri374 — COMMENTED** [review]
Thanks for this contribution! I added a small question.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py (hunk 1 above)
Comment (bckamil) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py:21]: The checksum calculation is described here: https://en.wikipedia.org/wiki/PESEL#Format.

It also describes the structure of PESEL (`YYMMDDZZZXQ`) and method to handle different centuries. You can verify my regular expression using this section.

Here you can generate some examples: https://pesel.cstudios.pl/O-generatorze/Generator-On-Line

Comment (omri374) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py:21]: Thanks for clarifying. I generated a few examples and have some questions:
1. Why are there `^` on the first regex and `$` on the second? A sentence could have a PESEL number in the middle too.
2. Looks like both these regexes do not catch the entire span of the pesel number. For instance, the second regex looks for 2 digits (starting with 0, 1, 2 or 3) + 5 digits = 7 digits, but pesel has more digits (11)

File: presidio-analyzer/tests/test_pl_pesel_recognizer.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    return ["PL_PESEL"]
+
+
+@pytest.mark.parametrize(
+    "text, expected_len, expected_positions",
+    [
+        # fmt: off
+        # valid PESEL scores
+        ("11111111114", 1, ((0, 11),),),
+        # invalid PESEL scores
```
Comment (omri374) [inline presidio-analyzer/tests/test_pl_pesel_recognizer.py:23]: Please add cases where the pesel number is in the middle of the sentence (e.g. `my pesel is 11111111114`)

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    :param context: List of context words to increase confidence in detection
+    :param supported_language: Language this recognizer supports
+    :param supported_entity: The entity this recognizer can detect
+    """
+
+    PATTERNS = [
+        Pattern(
+            "PESEL",
+            r"[0-9]{2}([02468][1-9]|[13579][012])(0[1-9]|1[0-9]|2[0-9]|3[01])[0-9]{5}",
+            0.5,
```
Comment (omri374) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py:22]: Can we consider reducing the default confidence to 0.4? While I don't anticipate this regex to create too many false positives, the default threshold is usually 0.4. With context words and/or checksum, this score will increase. WDYT?

Comment (bckamil) [reply presidio-analyzer/presidio_analyzer/predefined_recognizers/pl_pesel_recognizer.py]: Ok

**Review by omri374 — APPROVED** [review]

**Pull request merged**